platform-sdk/swirlds-common/src/testFixtures/java/com/swirlds/common/test/fixtures/Randotron.java

```diff
@@ -0,0 +1,197 @@
+import java.time.Instant;
+import java.util.Random;
+
+/**
+ * An extension of the Random class that provides additional utility methods for generating random data.
+ */
+public final class Randotron extends Random {
+    private final long seed;
+
+    /**
+     * Create a new instance of Randotron
+     *
+     * @return a new instance of Randotron
+     */
+    public static Randotron create() {
+        return new Randotron(new Random().nextLong());
+    }
+
+    /**
+     * Create a new instance of Randotron with the given seed
+     *
+     * @param seed the seed to use
+     * @return a new instance of Randotron
+     */
+    public static Randotron create(final long seed) {
+        return new Randotron(seed);
+    }
+
+    /**
+     * The ONLY permitted constructor.
+     * <p>
+     * Do NOT implement an unseeded constructor.
+     *
+     * @param seed the random seed
+     */
+    private Randotron(final long seed) {
+        super(seed);
+        this.seed = seed;
+        System.out.println("Random seed: " + seed + "L");
+    }
+
+    /**
+     * Generates a random string of the given length.
+     *
+     * @param length the length of the string to generate
+     * @return a random string
+     */
+    @NonNull
+    public String randomString(final int length) {
+        final int LEFT_LIMIT = 48; // numeral '0'
+        final int RIGHT_LIMIT = 122; // letter 'z'
+
+        return this.ints(LEFT_LIMIT, RIGHT_LIMIT + 1)
+                .filter(i -> (i <= 57 || i >= 65) && (i <= 90 || i >= 97))
+                .limit(length)
+                .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
+                .toString();
+    }
+
+    /**
+     * Generates a random IP address
+     *
+     * @return a random IP address
+     */
+    @NonNull
+    public String randomIp() {
+        return this.nextInt(256) + "." + this.nextInt(256) + "." + this.nextInt(256) + "." + this.nextInt(256);
+    }
+
+    /**
+     * Generates a random positive long that is smaller than the supplied value
+     *
+     * @param maxValue the upper bound, the returned value will be smaller than this
+     * @return the random long
+     */
+    public long randomPositiveLong(final long maxValue) {
+        return this.longs(1, 1, maxValue).findFirst().orElseThrow();
+    }
+
+    /**
+     * Generates a random positive long
+     *
+     * @return the random long
+     */
+    public long randomPositiveLong() {
+        return randomPositiveLong(Long.MAX_VALUE);
+    }
+
+    /**
+     * Generates a random hash
+     *
+     * @return a random hash
+     */
+    @NonNull
+    public Hash randomHash() {
+        return new Hash(randomByteArray(DigestType.SHA_384.digestLength()), DigestType.SHA_384);
+    }
+
+    /**
+     * Generates Bytes with random data that is the same length as a SHA-384 hash
+     *
+     * @return random Bytes
+     */
+    public Bytes randomHashBytes() {
```

Comment:
nonNull missing, sorry just spotted this one

Reply:
fixed